### PR TITLE
fix(otel): Update jaeger config

### PR DESCRIPTION
**Pull request by Chickensoupwithrice**
### Description

Resolves [REL-171](https://linear.app/sourcegraph/issue/REL-171/cve-2024-36129-deploy-sourcegraph) for this repo

### Checklist

- [ ] [CHANGELOG.md](https://github.com/sourcegraph/sourcegraph/blob/main/CHANGELOG.md) updated
- [ ] [K8s Upgrade notes updated](https://github.com/sourcegraph/sourcegraph/blob/main/doc/admin/updates/kubernetes.md)
- [x] Sister [deploy-sourcegraph-k8s](https://github.com/sourcegraph/deploy-sourcegraph-k8s) change: https://github.com/sourcegraph/deploy-sourcegraph-k8s/pull/153
- [x] Sister [deploy-sourcegraph-docker](https://github.com/sourcegraph/deploy-sourcegraph-docker) change: https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/1063
- [ ] All images have a valid tag and SHA256 sum
- [x] I acknowledge that [deploy-sourcegraph-k8s](https://github.com/sourcegraph/deploy-sourcegraph-k8s) is now the preferred Kubernetes deployment repository

### Test plan

Manual testing

### CODE REVIEW

File: overlays/jaeger/jaeger.Deployment.yaml [hunk 1]
```diff
@@ -45,6 +45,10 @@ spec:
               protocol: TCP
             - containerPort: 14250
               protocol: TCP
+            - containerPort: 4320
+              protocol: TCP
+            - containerPort: 4320
+              protocol: TCP
```
Comment (bobheadxi) [inline overlays/jaeger/jaeger.Deployment.yaml:51]: duplicate ports?

Comment (bobheadxi) [reply overlays/jaeger/jaeger.Deployment.yaml:51]: Maybe we can choose just to expose one (the gRPC one)

Comment (Chickensoupwithrice) [reply overlays/jaeger/jaeger.Deployment.yaml:51]: Great catch!

File: overlays/jaeger/otel-collector.Deployment.yaml [hunk 2]
```diff
@@ -13,3 +13,5 @@ spec:
           env:
             - name: JAEGER_HOST
               value: jaeger-collector
+            - name: JAEGER_OTLP_GRPC_PORT
+              value: 4320
```
Comment (DaedalusG) [inline overlays/jaeger/otel-collector.Deployment.yaml:17]: Are we missing `- 'JAEGER_OTLP_HTTP_PORT=4321'`?

Comment (Chickensoupwithrice) [reply overlays/jaeger/otel-collector.Deployment.yaml:17]: We absolutely are, great catch!

**Review by bobheadxi — APPROVED** [review]

**Pull request merged**